CONTRIBUTING.md

```diff
@@ -35,10 +35,34 @@ There is also an atom plugin https://atom.io/packages/atom-cfn-lint.
 
 Always write templates in YAML. CloudFormation templates can be hard to parse. Always use clear, verbose naming and add inline comments when appropriate. Group resources by their function, not by their type. Add the filename at the top of the file as a comment (this makes it easier to identify which template was used to launch a stack).
 
-
-
-
-
+#### Function syntax
+
+Always prefer the short form syntax (YAML tags) of functions, except in the following situations:
+
+1. When function nesting does not support short form, such as using `Fn::ImportValue` with `Fn::Join`.
+
+2. When it would lead to the function being the only thing on the line:
+
+```yaml
+# Undesirable
+Pets:
+  - Animal: Dog
+    Name: Scraps
+  - !If
```

Comment:
Hrmm... but why can't you do this?  Is it just not allowed by CFN?

Reply:
It is allowed I just think it is harder to read. I find that having some actual value, like the string `Fn::If` on a line makes a lot more sense than a YAML tag which doesn't _really_ exist